libdrgn/python/language.c

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 - Omar Sandoval
+// SPDX-License-Identifier: GPL-3.0+
+
+#include "drgnpy.h"
+#include "../language.h"
+
+static PyObject *Language_repr(Language *self)
+{
+	return PyUnicode_FromFormat("Language.%s", self->attr_name);
+}
+
+static PyObject *Language_get_name(Language *self, void *arg)
+{
+	return PyUnicode_FromString(self->language->name);
+}
+
+static PyGetSetDef Language_getset[] = {
+	{"name", (getter)Language_get_name, NULL, drgn_Language_name_DOC, NULL},
+	{},
+};
+
+PyTypeObject Language_type = {
+	PyVarObject_HEAD_INIT(NULL, 0)
+	.tp_name = "_drgn.Language",
+	.tp_basicsize = sizeof(Language),
+	.tp_repr = (reprfunc)Language_repr,
+	.tp_flags = Py_TPFLAGS_DEFAULT,
+	.tp_doc = drgn_Language_DOC,
+	.tp_getset = Language_getset,
+};
+
+static PyObject *languages_py[DRGN_NUM_LANGUAGES];
+
+PyObject *Language_wrap(const struct drgn_language *language)
+{
+	PyObject *obj = languages_py[language - drgn_languages];
+	Py_INCREF(obj);
```

Comment:
Heads up, I changed this from your commit. Without this, I was getting crashes in one of the tests I added (but I wasn't able to repro in normal debugging). Increasing the ref once when adding it to the array below didn't seem to help.

I'm not sure if this is correct though, I'm still not sure how lifecycle is handled (especially on init and handing out like this).

Reply:
Yup this is correct, good catch. Whoever gets this is going to drop a reference when it's done with it, so we need to account for that. This fix should be folded into my commit.